### DISCUSSION TITLE
How to install docker-cli in code-server container?

### DISCUSSION THREAD

**Discussion by nhvu1988** [coder/code-server#1848] [Q&A] [answered]
In the [Docker in code-server container](https://github.com/cdr/code-server/blob/master/doc/FAQ.md#docker-in-code-server-container) says

> Install the docker CLI in the code-server container and you should be able to access the daemon!

But I'm not sure how to install the docker CLI in the code-server container. I have tried many ways to do but did not succeed, could you please help and update the documentation as well. 

Thank you!

**Comment by code-asher** [comment]
You should be able to install it following the [Docker install docs](https://docs.docker.com/engine/install/debian/#installation-methods).

You only need to install `docker-ce-cli` if the daemon is running outside the container.

**Comment by nhvu1988** [comment]
I have followed the guide but there's an issue on this command

```
root@ae5e1fd829e7:/home/smarthome# sudo add-apt-repository \
>    "deb [arch=armhf] https://download.docker.com/linux/debian \
>    $(lsb_release -cs) \
>    stable"
Hit:1 https://deb.nodesource.com/node_12.x bionic InRelease
Hit:2 https://dl.yarnpkg.com/debian stable InRelease                                                                                   
Ign:3 https://download.docker.com/linux/debian bionic InRelease                                                                        
Hit:4 http://ports.ubuntu.com/ubuntu-ports bionic InRelease                                         
Hit:5 http://ports.ubuntu.com/ubuntu-ports bionic-updates InRelease      
Err:6 https://download.docker.com/linux/debian bionic Release           
  404  Not Found [IP: 54.192.150.13 443]
Hit:7 http://ports.ubuntu.com/ubuntu-ports bionic-backports InRelease   
Hit:8 http://ports.ubuntu.com/ubuntu-ports bionic-security InRelease
Reading package lists... Done                      
E: The repository 'https://download.docker.com/linux/debian bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

So the next install `docker-ce-cli` is failure

```
root@ae5e1fd829e7:/home/smarthome# apt-get install docker-ce-cli
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package docker-ce-cli
```

I'm setting it up on pi4 and docker on pi running fine. Do you know how I could fix it? Thank you

**Comment by code-asher** [comment, marked as answer]
Oh isn't "bionic" Ubuntu? You might need to replace `debian` with `ubuntu` in that command. My bad, I gave you a link to the Debian install docs since our code-server container uses Debian "buster".

Ubuntu Docker install docs: https://docs.docker.com/engine/install/ubuntu/

**Comment by nhvu1988** [comment]
@code-asher perfect, it works now. I'm using code-server image for arm32v7 from linuxserver so maybe they use Ubuntu?

Another issue is I have tried to install docker-compose from [here](https://docs.docker.com/compose/install/)

```
sudo curl -L "https://github.com/docker/compose/releases/download/1.26.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
sudo chmod +x /usr/local/bin/docker-compose
```
optional but I tried as well
```
sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
```
 but it says
```
root@7e1c5d9846cd:/home/smarthome# docker-compose --version
/usr/local/bin/docker-compose: line 1: Not: command not found
```

Do you know how to make it work? Thank you

UPDATED - it works with `apt install docker-compose`

**Comment by code-asher** [comment]
Awesome! I'm glad you got it working.
